### ISSUE TITLE
disallow _from and _to in g.vs.select() in undirected graphs

### ISSUE THREAD

**Issue by yunesj** [igraph/python-igraph#103]
I was surprised by this behavior. If it can't be modified, then maybe a note in the documentation would help?

For an undirected graph...
```
assert g.vs.degree() == g.vs.indegree() + g.vs.outdegree()
```
raises `AssertionError`.

Instead, `g.vs.degree() == g.vs.indegree() == g.vs.outdegree()`.

Continuing on...
```
assert len(g.es.select(_from=1)) == g.vs[1].degree()
assert len(g.es.select(_from=1)) == g.vs[1].outdegree()
```
both raise an `AssertionError`.

What do you think?

**Comment by yunesj** [comment]
Also, this doesn't match the behavior suggested by the tutorial.

From the [tutorial](http://igraph.org/python/doc/tutorial/tutorial.html):
> Using `_target` or `_to` filters based on the target vertices. This is different from `_source` and `_from` if the graph is directed.

This implies that `_target` or `_to` are *not* different from `_source` and `_from` if the graph is undirected.
```
>>> g.is_directed()
False
>>> assert list(g.es.select(_target=2)) == list(g.es.select(_source=2))
```
raises `AssertionError`.

**Comment by crsqq** [comment]
From a theoretical standpoint, a distinction between indegree and outdegree in an undirected graph does not make sense.
However, if we want to talk about in- and outdegree in case of undirected graphs, there is only one implementation that makes sense: 
```
g.vs.degree() == g.vs.indegree() == g.vs.outdegree().
```
This is what most people would expect if they want to use the in- or outdegree of an undirected graph.


Regarding your second observation:

> Continuing on...
> assert len(g.es.select(_from=1)) == g.vs[1].degree()
> assert len(g.es.select(_from=1)) == g.vs[1].outdegree()
> both raise an AssertionError.

At first, this behavior seems strange.
However:
```
sum([len(g.es.select(_from=i)) for i in range(g.vcount())]) == g.ecount()
```
If you iterate through all nodes and select the edges that leave that node, you will get all edges in the graph once, not twice. Which is an interesting design decision. One could argue that one would expect the receive all adjacent edges for all nodes in every step.

**Comment by ntamas** [comment]
`assert len(g.es.select(_from=1)) == g.vs[1].degree()` raises an `AssertionError` because the way `_from` and `_to` are implemented.

Internally, the edges of an undirected graph are stored in a list such that each edge appears only once in the list (i.e. if there is an edge between vertices 2 and 4, it is recorded in one direction only but not the other). Furthermore, each edge has a `from` and a `to` attribute that is assigned in a way consistent with the direction that igraph has picked internally. The `_from` and `_to` selectors simply use this internal attribute from the edge, so they won't work consistently on undirected graphs. I know that this is confusing, but unfortunately I don't have too much spare time to dedicate to the maintenance of `python-igraph`, so I cannot tell yet when and how this will be fixed - unless someone proposes a solution and sends me a pull request.

**Comment by ntamas** [comment]
This is actually a duplicate of #30 now that it's formulated this way.
